src/parser/mod.rs

```diff
@@ -9852,6 +9852,8 @@ impl<'a> Parser<'a> {
                     if self.parse_keyword(Keyword::UNSIGNED) {
                         Ok(DataType::TinyIntUnsigned(optional_precision?))
                     } else {
+                        let _ = dialect.allow_optional_signed_suffix()
+                            && self.parse_keyword(Keyword::SIGNED);
```

Comment:
```suggestion
                        if dialect.allow_optional_signed_suffix() {
                            let _ = self.parse_keyword(Keyword::SIGNED);
                        }
```
Thinking we can be explicit with an if statement for these?